### PR TITLE
Memory Allocation in Operating Systems in Java

### PR DESCRIPTION
**Pull request by shraiyya**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

PR corresponding to issue #1984 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Please add the DWOC tag and level of the code.

### CODE REVIEW

File: Java/Operating-System/alloc.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
+/* 
+Title: Memory Allocation in Operating Systems in Java
+
+Description:
+(a) First-Fit: Allocate the first hole that is big enough. Searching can start either at the beginning of the set of holes or where the previous first search ended. We can stop searching as soon as we find a free hole that is large enough.
```
Comment (HarshCasper) [inline Java/Operating-System/alloc.java]: Break down the line into shorter, more consumable lines.

File: Java/Operating-System/alloc.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+(b) Best-Fit: Allocate the smallest hole that is big enough. We must search the entire list unless the list is ordered by size.
+(c) Worst-Fit: Allocate the largest hole. We must search the entire list unless the list is ordered by size.
+ 
+*/
+
+import java.util.ArrayList;
+import java.util.Scanner;
+public class alloc {
+
+  public static void main(String[] args) {
```
Comment (HarshCasper) [inline Java/Operating-System/alloc.java]: Keep your main() function only for the Driver Code. Break down your entire code into modular functions and then call them from the main() function

File: Java/Operating-System/alloc.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+          else System.out.print("Not Allocated");
+          System.out.println();
+        }       
+        break;
+      }
+    } while ( u != 0 );
+
+  }
+
+}
```
Comment (HarshCasper) [inline Java/Operating-System/alloc.java]: Add sample I/O in a Multi-Line Comment

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Comment by shraiyya** [comment]
Please add the level of the code.

**Pull request merged**